buddy_alloc.h

```diff
@@ -917,16 +1015,19 @@ static struct buddy_tree_pos deepest_position_for_offset(struct buddy *buddy, si
 }
 
 static struct buddy_tree_pos position_for_address(struct buddy *buddy, const unsigned char *addr) {
+    struct buddy_tree *tree;
+    struct buddy_tree_pos pos;
+
     /* Find the deepest position tracking this address */
     unsigned char *main = buddy_main(buddy);
     ptrdiff_t offset = addr - main;
 
-    if (offset % buddy->alignment) {
+    if (offset % (ptrdiff_t)buddy->alignment) {
         return INVALID_POS; /* invalid alignment */
     }
 
-    struct buddy_tree *tree = buddy_tree(buddy);
-    struct buddy_tree_pos pos = deepest_position_for_offset(buddy, offset);
+    tree = buddy_tree(buddy);
+    pos = deepest_position_for_offset(buddy, (size_t)offset);
```

Comment:
offset is ptrdiff_t. Is there an assumption that offset would be always positive? Otherwise, the implicit conversion seems to be a bug.

Reply:
The offset calculated this way for a slot within the allocator arena is always 0 or positive.